feugy/tabulous
# feat(web): adds terms of services

### :book: What's in there?

Let's get our b..t covered a little bit. This Terms of Service pages covers legal mentions, Terms of Service and Cookie Policy.
Because it must be reachable, we need a footer with a link to it.
Because some page (connected page, Terms of Services) could be tall, we need the header to be sticky. And the footer to have a "scroll to top" button.

The Terms of Service page comes in two flavors: as a standalone page (with legal data), and as part of the first connection process. After logging in for the first time, players will have to scroll through the terms, and check the 2 checkboxes to access Tabulous. If they don't, they'll be blocked on this page.

Are included here:

- fix: dev console output is truncated since using Pnpm
- fix(server): graphiQL url returns a 404 in dev mode
- fix(server): redis DB is cleared after test run
- fix(cli): no help displayed when granting an unknown player
- feat(web): adds terms of service + legal standalone page
- feat(web): introduces accept-terms page for first connection, which redirect to original page
- feat(web): includes a page footer in all pages
- feat(web): makes page header sticky
- feat(server): introduces mutation to accept terms
- feat(cli): introduces show-player command
- refactor(web): moves login form component in its page folder

### :test_tube: How to test?

The e2e test suite for home page covers displaying, scrolling and accepting the Terms of Service.
You can try it live:
1. pnpm start
2. log with any account
   > you'll be shown the accept terms page, that you have to scroll, and accept, before continuing
3. refresh, or logs out and in
   > you should not see the accept terms page any more.
4. click _Legal & Terms of Service_ link in the page footer
   > you'll browse to a page showing both legal and Terms of Service, with breadcrumb
